src/main/java/ca/on/oicr/pineryreports/reports/impl/PreciseCaseReport.java

```diff
@@ -0,0 +1,229 @@
+     * tracking tissue origin is critical.
+     */
+    public Predicate<SampleDto> predicate() {
+      return sample -> {
+        if (SAMPLE_CATEGORY_IDENTITY.equals(getAttribute(ATTR_CATEGORY, sample))) return false;
+        return timePointCode == getTimesReceived(sample.getName());
+      };
+    }
+  }
+
+  @Override
+  protected void collectData(PineryClient pinery) throws HttpResponseException, IOException {
+    List<SampleDto> allPreciseSamples =
+        pinery
+            .getSample()
+            .allFiltered(new SampleClient.SamplesFilter().withProjects(Lists.newArrayList("PRE")));
+    Map<String, SampleDto> allPreciseSamplesById = mapSamplesById(allPreciseSamples);
+    Map<String, List<SampleDto>> allPreciseSamplesByIdentity =
+        mapChildrenByIdentityId(allPreciseSamplesById);
+    List<String> tissueOriginOrder = Lists.newArrayList("Se", "Pl", "Ly");
+    List<String> tissueOriginOrderExtra = Lists.newArrayList("Se", "Pl", "Ly", "Us", "Up");
+    allPreciseIdentities =
+        allPreciseSamples
+            .stream()
+            .filter(s -> getAttribute(ATTR_CATEGORY, s).equals(SAMPLE_CATEGORY_IDENTITY))
+            .collect(Collectors.toList());
+
+    table = new LinkedList<>();
+
+    // One row per identity
+    for (SampleDto identity : allPreciseIdentities) {
+      List<String> row = new LinkedList<>();
+      row.add(identity.getName());
+
+      // For all regular time points
+      for (int i = 0; i < 5; i++) {
+        TimePoint currentTimePoint = TimePoint.values()[i];
+        List<SampleDto> samplesAtCurrentTime =
+            allPreciseSamplesByIdentity
+                .get(identity) // Get children of this identity
+                .stream()
+                .filter(currentTimePoint.predicate()) // Get samples at this time point
+                .collect(Collectors.toList());
+        // on 0th, 2th, 4th iterations use all the types, otherwise just 3
+        List<String> tissueOrigins;
+        if (i % 2 == 0) {
+          tissueOrigins = tissueOriginOrderExtra;
+        } else {
+          tissueOrigins = tissueOriginOrder;
+        }
+
+        // for all applicable tissue origins, get counts and add them to the row
+        for (String origin : tissueOrigins) {
+          row.add(
+              String.valueOf(
+                  samplesAtCurrentTime
+                      .stream()
+                      .filter(s -> origin.equals(getAttribute(ATTR_TISSUE_ORIGIN, s)))
+                      .count()));
+        }
+      }
+
+      // Handle irregular time points
+      row.add(
+          String.valueOf(
+              allPreciseSamplesByIdentity
+                  .get(identity) // Get children of this identity
+                  .stream()
+                  .filter(s -> s.getSampleType().equals(SAMPLE_CLASS_SLIDE))
+                  .filter(TimePoint.BX.predicate())
+                  .filter(s -> getAttribute(ATTR_GROUP_ID, s).equals("POS"))
```

Comment:
Group ID will only be present on the sample it's set on, but it should affect children as well, so parents should be checked using `SampleUtils#getUpstreamAttribute` too

Reply:
Updated